### PR TITLE
[AIRFLOW-5518] Use correct scheme for Http connection parsed from URI

### PR DESCRIPTION
**Pull request by K3UL**
[AIRFLOW-5518] Add tests for cases of HTTPS urls

Make sure you have checked _all_ steps below.

### Jira

- [X] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5518

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:
If a connection is parsed from a URI (for example if it's overloaded through an environment variable), the schema is the path or the URI.

This is wrong because then the HttpHook uses the schema to prefix the base_url. It is particularly visible when using https, which gets transformed to default http. As a result the URLs are built with HTTP instead of HTTPS.

I propose the following, which it is consistent with what happens when a connection is added through other ways (UI and config files):
At the Connection level, in parse_from_uri(), implement a special treatment for http connection to have conn_type='http' and schema=scheme

### Tests

- [X] My PR adds the following unit tests :
3 new tests in test_connection_from_uri_with_path, targeting HTTPS. I also changed the pre-existing tests with HTTP URIs to indicate that the schema is HTTP.

### Commits

- [X] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [X] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: tests/models/test_connection.py [hunk 1]
```diff
@@ -212,31 +212,31 @@ def test_connection_from_uri_with_path(self):
             (
                 "http://:password@host:80/database",
                 ConnectionParts(
-                    conn_type="http", login='', password="password", host="host", port=80, schema="database"
+                    conn_type="http", login='', password="password", host="host", port=80, schema="http"
```
Comment (ashb) [inline tests/models/test_connection.py:215]: I am not a fan of this - it papers over the bug and makes it more confusing - schema does not mean the same as scheme.

Comment (K3UL) [reply tests/models/test_connection.py:215]: While I agree that schema does not equal scheme, it is what is used by the HttpHook, and it is where HTTP/HTTPS goes if you put it through the UI.
As you mentioned in the Jira, this should be refactored deeper, but IMO this fix brings to the parser a behavior that is consistent with what happens through the UI.

The alternative is to edit the HttpHook to change `schema = conn.schema if conn.schema else "http"` with `schema = conn.schema or conn.conn_type or 'http'`.
I'm open to it if you think it's better, but I think it's worse, as it invents a new Connection type (https) that is not referenced in `_types`.

**Comment by mik-laj** [comment]
Can you look at this PR? 
https://github.com/apache/airflow/pull/5239
I once started working on it, but I didn't find time to finish it.

**Comment by K3UL** [comment]
@mik-laj Ah ! I had looked into it but didn't find this one ; From what I see, the only thing missing is the use of schema to know if it's http or https.
How would you like to proceed, for me to review it or continue working on your PR ? I'm not sure how one does that, I'm a little new to GitHub workflows.

**Comment by mik-laj** [comment]
@K3UL In my opinion, it will be more valuable if you finish working on my PR. This PR solves more problems. Are you interested in this? I don't have time to finish my PR, but I will gladly review your changes if you create a PR to my PR. Do you like this solution?

**Pull request closed without merging**